### DISCUSSION TITLE
Behaviour after power outage

### DISCUSSION THREAD

**Discussion by guydvir2** [arendst/Tasmota#22459] [Q&A]
I'm using a Athom wall switch ( for the discussion 4 gang ) and all Tasmotas using a local MQTT server, and HomeAssistant.
Some of my devices (~ 35 units) configured as curtains (always on Relay1 and Relay2).
After poweroutage, and power back on, all Curtains went UP exposing my home to any outside visitor.

I can go now for each device and re-wire it and change the code to relocate the shutter to Relay3 +Relay4, but it has no sense.

Is there a rule that can solve that security problem ?
When I tried ChatGPT: I did not get the desired behaviour. 
Rule3 ON System#Boot DO Var2 1; Power1 OFF; ENDON Rule3 ON Var2#State=1 DO Power1 OFF; Delay 2; Var2 0; ENDON

Guy

**Comment by sfromis** [comment]
Stop using ChatGPT here, unless you **want** broken hallucination to just confuse you more, like that weird contraption.

What to do depends on exactly how you want the relay states to be processed, depending on how your curtains work. Options to consider include using `PowerOnState` and `SetOption0`, and only possibly rules not like what ChatGPT mixed up.

**Comment by guydvir2** [comment]
Below, definition of a curtain :
{"SetOption0":"ON"}
{"PowerOnState":3}
backlog SetOption80 1;ShutterRelay1 1; Interlock 1,2; Interlock ON; 

My goal is avoid curtain goes up after power-up 

Guy

**Reply by sfromis** [reply]
But that depends on what exact behavior you expect on the relay pins to achieve that.
`PowerOnState 3` means "last saved", but that may not be how to control *your* curtains.

The [docs](https://tasmota.github.io/docs/Blinds-and-Shutters/#operation) (do read) suggests this, but I have zero experience with that feature.
> If possible to avoid any injury on unexpected movement all RELAYS should start in OFF mode when the device reboots: `PowerOnState 0`

**Reply by guydvir2** [reply]
@sfromis  _ Yes, I know that this is the meaning of  `PowerOnState`, but all curtains behave the same (going UP) when they all were last at "Stop" position.

For security sake I'd rather all relays (Lights) and Shutter init at Off (Lights) and Down(Shutters).
Is it achievable  ?

**Reply by guydvir2** [reply]
BTW, changing PowerOnState to 0, kept same behavior.

**Comment by guydvir2** [comment]
@sfromis 
If I cannot initiate "stop" of relevant Relay at boot, I rather have `ShutterPosition` 0 rather than an open curtain.
It it possible ?

**Reply by sfromis** [reply]
As I posted already, "I have zero experience with that feature"
